view/src/__tests__/Callstack.test.jsx

```diff
@@ -0,0 +1,129 @@
+// Unit tests of Callstack component
+// Author: Lukas Petr
+
+import {
+  fireEvent, render, screen,
+} from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import Callstack from '../components/Callstack';
+
+describe('simple callstack visualisation test', () => {
+  const compFunName = '__put_task_struct';
+  const oldCallStack = [{ name: 'free_task' }, { name: 'free_task_struct' }];
+  const newCallStack = [{ name: 'free_task' }, { name: 'free_task_struct' }];
+  const onSelect = jest.fn();
+
+  const setup = () => {
+    render(
+      <Callstack
+        compFunName={compFunName}
+        oldCallStack={oldCallStack}
+        newCallStack={newCallStack}
+        definitions={{}}
+        onSelect={onSelect}
+      />,
+    );
+  };
+  it('all function names should be visible', () => {
+    setup();
+    expect(screen.getByText('__put_task_struct')).toBeVisible();
+    expect(screen.getByText('free_task')).toBeVisible();
+    expect(screen.getByText('free_task_struct')).toBeVisible();
+  });
+  it('click on function should call onSelect callback', () => {
+    setup();
+    fireEvent.click(screen.getByText('__put_task_struct'));
+    expect(onSelect).toHaveBeenCalled();
+  });
+});
+
+describe('complex callstack visualisation test', () => {
```

Comment:
Could we somehow check if the call stacks are properly "aligned"? For example, that `percpu_from_op` calls are in the same row but also that the non-matching parts of the stacks are properly aligned?

I know that it's sometimes hard to test things in React so if that's too much effort, let's skip it for now and think about it after we merge this PR.

Reply:
I will look at it.

Reply:
I think that this is the last remaining thing on this PR. I'd suggest merging it as it is now and potentially add this test later.

Reply:
I added tests for testing call stack alignment in PR #274.